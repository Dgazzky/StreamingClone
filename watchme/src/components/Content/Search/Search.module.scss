@import '../../../constants/colors';
.search {
    display: flex;
    flex-direction: row;
    align-items: center;
    border: 2px solid $color-gray-100;
    border-radius: 20px;
    padding: 14px;
    [class='ant-input'] {
        background-color: transparent;
        outline: none;
        border: none;
        height: 20px;
        margin-left: 14px;
        font-weight: 600;

        &:focus {
            outline: none;
            border: none;
        }
    }
}
