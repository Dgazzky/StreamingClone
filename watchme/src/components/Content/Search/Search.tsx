import React from 'react'
import { Input } from 'antd'
import { SearchOutlined } from '@ant-design/icons'
import styles from './Search.module.scss'
import { COLORS } from '../../../constants/colors'
import PropTypes from 'prop-types'

const Search = (props: any) => {
    const { placeholder, value, onChange } = props
    return (
        <div className={styles.search}>
            <Input
                placeholder={placeholder}
                value={value}
                onChange={onChange}
                prefix={<SearchOutlined style={{ color: COLORS['gray'] }} />}
            />
        </div>
    )
}
Search.propTypes = {
    placeholder: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    searchString: PropTypes.string.isRequired,
}

export default Search
