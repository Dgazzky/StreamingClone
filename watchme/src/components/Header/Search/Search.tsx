import React, { useState } from 'react'
import { Input } from 'antd'
import styles from './Input.modules.scss'

interface SearchProps {
    onSearch: (value: string) => void
}
const Search = ({ onSearch }: SearchProps) => {
    const [searchString, setSearchString] = useState('')
    return (
        <div className={styles.searchContainer}>
            <Input.Search
                value={searchString}
                placeholder={'Search'}
                onChange={event => setSearchString(event.target.value)}
                onSearch={onSearch}
            />
        </div>
    )
}
export default Search
