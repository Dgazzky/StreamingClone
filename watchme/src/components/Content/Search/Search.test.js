import React from 'react'

import Adapter from 'enzyme-adapter-react-16'
import Search from './Search'
import { shallow, configure } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

configure({ adapter: new Adapter() })
const mockFunction = jest.fn()

describe('Search', function () {
    it('renders correctly', () => {
        const component = shallow(
            <Search onChange={mockFunction} searchString={''} />
        )
        expect(shallowToJson(component)).toMatchSnapshot()
    })
    it('changes search value', () => {
        const component = shallow(
            <Search onChange={mockFunction} searchString={'test'} />
        )
        expect(shallowToJson(component)).toMatchSnapshot()
    })
})
