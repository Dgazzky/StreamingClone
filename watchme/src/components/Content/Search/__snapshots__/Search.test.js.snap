// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search changes search value 1`] = `
<div
  className="search"
>
  <Input
    onChange={[MockFunction]}
    prefix={
      <ForwardRef(SearchOutlined)
        style={
          Object {
            "color": "#A9AEC1",
          }
        }
      />
    }
    type="text"
  />
</div>
`;

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <Input
    onChange={[MockFunction]}
    prefix={
      <ForwardRef(SearchOutlined)
        style={
          Object {
            "color": "#A9AEC1",
          }
        }
      />
    }
    type="text"
  />
</div>
`;
